src/test/java/org/sonatype/nexus/ci/nxrm/v3/CreateTagStepTest.groovy

```diff
@@ -0,0 +1,350 @@
+      project.setDefinition(new CpsFlowDefinition("""
+          node {
+            def file = new File("\${WORKSPACE}/${attrFileName}")
+            file.createNewFile()
+            file << '{"foo": "bar"}'
+            createTag nexusInstanceId: '${instance}', tagName: '${tagName}', tagAttributesPath: '${attrFileName}'
+          }
+          """))
+
+      GroovyMock(RepositoryManagerClientUtil.class, global: true)
+      RepositoryManagerClientUtil.nexus3Client(instance) >> nxrm3Client
+      RepositoryManagerClientUtil.nexus3Client(config.serverUrl, config.credentialsId) >> nxrm3Client
+
+    when:
+      def build = project.scheduleBuild2(0).get()
+
+    then:
+      1 * nxrm3Client.createTag(tagName, [foo: 'bar']) >> new Tag(tagName, [foo: 'bar'])
+      jenkins.assertBuildStatus(Result.SUCCESS, build)
+  }
+
+  @Unroll
+  def 'tag attributes are optional - #description'(tagName, attributesFile, attributesJson, attributeMap,
+                                                   description)
+  {
+    setup:
+      def job = prepareJob('nx3', tagName)
+
+      if (attributesFile) {
+        def attrFile = Paths.get(job.workspace.absolutePath, 'attr-file.json')
+        Files.write(attrFile, attributesFile.bytes, StandardOpenOption.CREATE_NEW)
+        job.builder.setTagAttributesPath('attr-file.json')
+      }
+
+      if (attributesJson) {
+        job.builder.setTagAttributesJson(attributesJson)
+      }
+
+    when:
+      def build = job.project.scheduleBuild2(0).get()
+
+    then:
+      1 * nxrm3Client.createTag(tagName, attributeMap) >> new Tag(tagName, attributeMap ?: [:])
+      jenkins.assertBuildStatus(Result.SUCCESS, build)
+
+    where:
+      tagName << ['tagFoo', 'tagBar', 'tagBaz']
+      attributesFile << [null, '{"baz": "qux"}', null]
+      attributesJson << ['{"foo": "bar"}', null, null]
+      attributeMap << [[foo: 'bar'], [baz: 'qux'], null]
+      description << ['no attribute file', 'no attribute json', 'no attributes']
+  }
+
+  def 'tag attribute json has priority over file attributes'() {
+    setup:
+      def tagName = 'create-tag-test'
+      def job = prepareJob('nx3', tagName)
+      def attrFile = Paths.get(job.workspace.absolutePath, 'attr-file.json')
+      Files.write(attrFile, '{"foo": "bar", "baz": "qux"}'.bytes,
+          StandardOpenOption.CREATE_NEW) // create file in workspace
+
+      job.builder.setTagAttributesPath('attr-file.json') // uses relative path in workspace
+      job.builder.setTagAttributesJson('{"foo": "quux"}')
+
+    when:
+      def build = job.project.scheduleBuild2(0).get()
+
+    then:
+      1 * nxrm3Client.createTag(tagName, [foo: 'quux', baz: 'qux']) >> new Tag(tagName, [foo: 'quux', baz: 'qux'])
+      jenkins.assertBuildStatus(Result.SUCCESS, build)
+  }
+
+  @Unroll
+  def 'descriptor validates tag name - #description'(tagName, validationKind, validationMessage, description) {
+    setup:
```

Comment:
Believe this test need to be updated to remove the tests that previously exercised the validation that has been removed. Running locally it looks like the last 4 tests failed for me.

Reply:
Yes, thanks. Forgot to cleanup before pushing a commit out last night before calling it a day :)

Reply:
Should be resolved in https://github.com/sonatype/nexus-platform-plugin/pull/16/commits/ad0b287bafeef49a9945f8cffb49a92a4e7cadbc